ov_msckf/src/slam2.cpp

```diff
@@ -205,29 +206,40 @@ class slam2 : public plugin {
 
 	virtual void start() override {
 		plugin::start();
-		sb->schedule<imu_cam_type>(id, "imu_cam", [&](switchboard::ptr<const imu_cam_type> datum, std::size_t iteration_no) {
+		sb->schedule<imu_type>(id, "imu", [&](switchboard::ptr<const imu_type> datum, std::size_t iteration_no) {
 			this->feed_imu_cam(datum, iteration_no);
 		});
 	}
 
 
-	void feed_imu_cam(switchboard::ptr<const imu_cam_type> datum, std::size_t iteration_no) {
+	void feed_imu_cam(switchboard::ptr<const imu_type> datum, std::size_t iteration_no) {
 		// Ensures that slam doesnt start before valid IMU readings come in
-		if (datum == NULL) {
+		if (datum == nullptr) {
 			return;
 		}
 
 		// Feed the IMU measurement. There should always be IMU data in each call to feed_imu_cam
-		assert((datum->img0.has_value() && datum->img1.has_value()) || (!datum->img0.has_value() && !datum->img1.has_value()));
-		open_vins_estimator.feed_measurement_imu(duration2double(datum->time.time_since_epoch()), datum->angular_v.cast<double>(), datum->linear_a.cast<double>());
+		open_vins_estimator.feed_measurement_imu(duration2double(datum->time.time_since_epoch()), datum->angular_v, datum->linear_a);
 
+		switchboard::ptr<const cam_type> cam;
+		// Buffered Async:
+		cam = _m_cam.size() == 0 ? nullptr : _m_cam.dequeue();
 		// If there is not cam data this func call, break early
-		if (!datum->img0.has_value() && !datum->img1.has_value()) {
+		if (!cam) {
 			return;
-		} else if (imu_cam_buffer == NULL) {
-			imu_cam_buffer = datum;
+		}
+		if (!cam_buffer) {
+			cam_buffer = cam;
+			return;
+		}
+		/* This is necsesary in async, to make sure that cam is new
+		   In buffered, cam is new if it is present.
```

Comment:
do we still need this

Reply:
Yes. We do not change anything in OpenVINS and the buffer is to be consistent with OpenVINS. We may not need it when we integrate the latest OpenVINS.

Reply:
I think it is better to have a flag for these. CAMERA_ASYNC something like that

Reply:
I think this comment is stale. We are not using async but buffered reader, which guarantees that if a cam is present, it is new. I'll remove the comment.

Reply:
ok thats what i thought. just remove it in this case.